uart/uart_posix.c

```diff
@@ -153,12 +153,12 @@ bool uart_receive(const serial_port sp, byte_t* pbtRx, size_t pszMaxRxLen, size_
     if (res < 0) {
       return false;
     }
- 
+
     // Read time-out
     if (res == 0) {
       if (*pszRxLen == 0) {
         // Error, we received no data
-        return false;
+        return true;
```

Comment:
Why does this path also return true? Don't you want to know when there is an error?

Reply:
Timing out without receiving any data may not be an error if the hardware just doesn't have any data ready for us at the moment.

Reply:
uart_posix.c does some additional and unnecessary re-packaging which is already handled in proxmark3.c `uart_receiver()`:
```
			if (prx-rx < sizeof(UsbCommand)) {
				continue;
			}
```
uart_win32.c doesn't do that. Why not aligning/simplyfying the Unix version and make it behave like the Windows version?

Reply:
Ah true. I also double checked `ReadFile` [(the win32api function that this is equivalent to)](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365467(v=vs.85).aspx), that doesn't return false on timeout.

So that entire `if` statement can go.

Reply:
The Win and *nix APIs are totally different.  It's been a while since I looked at uart_posix.c and I really didn't spend a huge amount of time in there, but from what I recall the various return's are to break out of a loop or to handle the various conditions the select() can result in (read error vs "no data available" which isn't an error).  If it's rewritten to be like uart_win32.c then it will block forever (or at least until the PM3 is unplugged) and not allow the comms thread to exit on client shutdown due to the "do { ... } while(byteCount);" .  The only clean-up I see is the "if(..) return true; else return true;"  As per my first comment I have absolutely no clue if the WIN32 routine returns an error on no data or not, or an error on a read error or not.  No windoze boxen in this building...

Reply:
FYI, if you want to test stuff on Windows, https://modern.ie/ has time-limited trial Windows virtual machines. They're ostensibly for testing websites in Internet Explorer and Edge, but you can install any software you like on them.

That should be sufficient for getting ProxSpace (the Windows build environment) going.